catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -845,15 +591,449 @@ function DropzoneMessage({ error, warn, disabled }: DropzoneMessageProps) {
+}
+
+const useHeaderStyles = M.makeStyles({
+  root: {
+    display: 'flex',
+    height: 24,
+  },
+})
+
+function Header(props: React.PropsWithChildren<{}>) {
+  const classes = useHeaderStyles()
+  return <div className={classes.root} {...props} />
+}
+
+const useHeaderTitleStyles = M.makeStyles((t) => ({
+  root: {
+    ...t.typography.body1,
+    alignItems: 'center',
+    display: 'flex',
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+  },
+  regular: {},
+  disabled: {
+    color: t.palette.text.secondary,
+  },
+  error: {
+    color: t.palette.error.main,
+  },
+  warn: {
+    color: t.palette.warning.dark,
+  },
+}))
+
+type HeaderTitleState = 'disabled' | 'error' | 'warn' | 'regular'
+
+function HeaderTitle({
+  state = 'regular',
+  ...props
+}: React.PropsWithChildren<{ state?: HeaderTitleState }>) {
+  const classes = useHeaderTitleStyles()
+  return <div className={cx(classes.root, classes[state])} {...props} />
+}
+
+const PROGRESS_EMPTY = { total: 0, loaded: 0, percent: 0 }
+
+const useLockStyles = M.makeStyles((t) => ({
+  root: {
+    alignItems: 'center',
+    background: 'rgba(255,255,255,0.9)',
+    border: `1px solid ${t.palette.action.disabled}`,
+    borderRadius: t.shape.borderRadius,
+    bottom: 0,
+    cursor: 'not-allowed',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressContainer: {
+    display: 'flex',
+    position: 'relative',
+  },
+  progressPercent: {
+    ...t.typography.h5,
+    alignItems: 'center',
+    bottom: 0,
+    display: 'flex',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressSize: {
+    ...t.typography.body2,
+    color: t.palette.text.secondary,
+    marginTop: t.spacing(1),
+  },
+}))
+
+function Lock({
+  progress,
+}: {
+  progress?: {
+    total: number
+    loaded: number
+    percent: number
+  }
+}) {
+  const classes = useLockStyles()
+  return (
+    <div className={classes.root}>
+      {!!progress && (
+        <>
+          <div className={classes.progressContainer}>
+            <M.CircularProgress
+              size={80}
+              value={progress.total ? progress.percent : undefined}
+              variant={progress.total ? 'determinate' : 'indeterminate'}
+            />
+            {!!progress.total && (
+              <div className={classes.progressPercent}>{progress.percent}%</div>
+            )}
+          </div>
+          {!!progress.total && (
+            <div className={classes.progressSize}>
+              {readableBytes(progress.loaded)}
+              {' / '}
+              {readableBytes(progress.total)}
+            </div>
+          )}
+        </>
+      )}
+    </div>
+  )
+}
+
+const useFilesContainerStyles = M.makeStyles((t) => ({
+  root: {
+    direction: 'rtl', // show the scrollbar on the left
```

Comment:
:(

Reply:
?

Reply:
It looks bad in my opinion. I hoped you would realize that after a while, but you stuck with this decision and scale it to another Dialog

Reply:
the main reason for that is scrollbar on macos blocking the buttons on the right from being clicked, it's not an aesthetics-based decision, but more like a workaround for a usability issue, and i'm open to alternatives if you can provide any.

it looks okayish imo since most of the time the scrollbar is hidden (on macos at least).

Reply:
The alternative is obvious, so I think you already declined it: show scrollbar at the right side, probably add some space with padding.

Reply:
padding on the right looks as bad as the scrollbar on the left)
one alternative that comes to mind is (ab)using horizontal scrollbar (akin to smth like telegram embedded reader), but this seems even more odd and unconventional for a primarily desktop-oriented (web)app